pkg/gateway/sig/javav2.go

```diff
@@ -0,0 +1,146 @@
+package sig
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"net/http"
+	"net/url"
+	"sort"
+	"strings"
+
+	"github.com/treeverse/lakefs/pkg/auth/model"
+	"github.com/treeverse/lakefs/pkg/gateway/errors"
+	"github.com/treeverse/lakefs/pkg/logging"
+)
+
+// Implements the "signing protocol" as exists at aws-sdk-java @ 1.12.390 (commit 07926f08a70).
+// This should never have worked, on any version of S3, but the implementation is there, unmodified since 2015.
+// The code is below, for reference
+/*
+
+private String calculateStringToSignV2(SignableRequest<?> request) throws SdkClientException {
```

Comment:
delete?

Reply:
No - this is a code snippet from the original java file